File: appsets-android-native/appsets/src/main/kotlin/xcj/app/appsets/db/room/dao/UserRelationDao.kt
package xcj.app.appsets.db.room.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import xcj.app.appsets.db.room.entity.UserRelation


@Dao
interface UserRelationDao {
    @Query("select * from userrelation")
    suspend fun getRelationList(): List<UserRelation>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun addUserRelation(vararg userRelation: UserRelation)
}

File: AppSets-iOS/AppSets/com/app/appsets/db/realm/dao/UserRelationDao.swift
//
//  UserRelationDao.swift
//  AppSets
//
//  Created by caiju Xu on 2024/7/13.
//

import Foundation

protocol UserRelationDao {
    
    func getRelationList() -> [UserRelation]
    
    func addUserRelation(_ realations: [UserRelation])
    
}
